### PR TITLE
[SUPTMTC-115] Accommodating the possibility of empty or '/' value for…

### PR DESCRIPTION
**Pull request by nku-ubiqube**
… the '$BASE_URL_MS' config variable

We need this fix to make our configurable capability to seamlessly work as currently it is hardcoded and thus do not work for thirdparty VNFM

### CODE REVIEW

**Review by UBiqube-ydu — APPROVED** [review]

File: adapters/vnfm_generic/polld/vnfm_generic_poll.php [hunk 1]
```diff
@@ -80,7 +80,13 @@ function get_health($PROTOCOL, $ME_IP, $HTTP_PORT, $BASE_URL_MS, $TOKEN)
   {
     global $health_path;
 
-    $url = "$PROTOCOL://$ME_IP:$HTTP_PORT/$BASE_URL_MS/$health_path";
+    //$url = "$PROTOCOL://$ME_IP:$HTTP_PORT/$BASE_URL_MS/$health_path";
+     $url = "$PROTOCOL://$ME_IP:$HTTP_PORT";
+    if (!(empty($BASE_URL_MS)) && ($BASE_URL_MS !== '/')){
+        $url .= "/$BASE_URL_MS";
```
Comment (yguubiqube) [inline adapters/vnfm_generic/polld/vnfm_generic_poll.php:86]: Can you ensure that BASE_URL_MS does not start or end with the / character?

**Review by yguubiqube — APPROVED** [review]

File: adapters/vnfm_generic/polld/vnfm_generic_poll.php (hunk 1 above)
Comment (abr-ubiqube) [reply adapters/vnfm_generic/polld/vnfm_generic_poll.php:86]: @nku-ubiqube 
can you adjust the code based on @yguubiqube request ?

File: adapters/vnfm_generic/polld/vnfm_generic_poll.php [hunk 2]
```diff
@@ -80,7 +80,13 @@ function get_health($PROTOCOL, $ME_IP, $HTTP_PORT, $BASE_URL_MS, $TOKEN)
...
+    }
```
Comment (yguubiqube) [inline adapters/vnfm_generic/polld/vnfm_generic_poll.php:87]: I suggest the replacement of lines 84 to 88 by:
```
$url = "$PROTOCOL://";
if (! isset($BASE_URL_MS)) $BASE_URL_MS = '/';
$url .= preg_replace("!/+!", "/", "$ME_IP:$HTTP_PORT/$BASE_URL_MS/$health_path");
```

**Review by yguubiqube — CHANGES_REQUESTED** [review]

File: adapters/vnfm_generic/polld/vnfm_generic_poll.php (hunk 1 above)
Comment (nku-ubiqube) [reply adapters/vnfm_generic/polld/vnfm_generic_poll.php]: Updated

**Review by yguubiqube — APPROVED** [review]

**Pull request merged**